shared/src/crypto.rs

```diff
@@ -333,6 +346,11 @@ impl CertificateCache {
                     continue;
                 }
             };
+            // Check if the new cert is already revoked
+            if certificate_revocation_list.as_ref().is_some_and(|list| is_revoked(list.get_by_cert(&opensslcert))) {
+                self.serial_to_x509.insert(serial.clone(), CertificateCacheEntry::Invalid(CertificateInvalidReason::Revoked));
+                continue;
```

Comment:
If I read the code correctly: This will prevent `new_count` from being increased. Thus, an update containing only revoked certificates (e.g. update of exactly 1 recovation) will yield `new_count==0`. Won't the calling function interpret this as "nothing has changed" and fail to update?